spinta/datasets/keymaps/redis.py

```diff
@@ -0,0 +1,181 @@
+            {
+                "cid": cid,
+                "updated": time.isoformat(),
+            }
+        )
+        self.redis.hset(self._get_sync_table_name(), key=name, value=data)
+
+    def synchronize(self, data: KeymapSyncData):
+        name = data.name
+        key = data.identifier
+        redirect = data.redirect
+        value = data.value
+        modified_at = data.data.get("_created")
+
+        value_table_name = self._get_value_table_name(name)
+        key_table_name = self._get_key_table_name(name)
+        metadata_table_name = self._get_metadata_table_name(name)
+
+        serialized_value = json.dumps(value, sort_keys=True)
+
+        self.redis.hset(value_table_name, serialized_value, key)
+        self.redis.hset(key_table_name, key, serialized_value)
+        existing_metadata = self.redis.hget(metadata_table_name, key)
+        metadata = json.loads(existing_metadata) if existing_metadata else {}
+        if modified_at:
+            metadata["modified_at"] = json.dumps(modified_at)
+        if redirect:
+            metadata["redirect"] = json.dumps(redirect)
+        if metadata:
+            self.redis.hset(metadata_table_name, key, json.dumps(metadata, sort_keys=True))
+
+    def validate_data(self, name: str):
+        value_table_name = self._get_value_table_name(name)
+        all_values = self.redis.hvals(value_table_name)
+        counts = Counter(all_values)
+        affected_key_count = sum(1 for v in counts.values() if v > 1)
+        if affected_key_count:
+            affected_row_count = sum(v - 1 for v in counts.values() if v > 1)
+            if self.duplicate_warn_only:
+                echo(
+                    (
+                        f"WARNING: Keymap's ({self.name!r}) {name!r} key contains {affected_key_count} duplicate value combinations.\n"
+                        f"This affects {affected_row_count} keymap entries.\n\n"
+                        "Make sure that synchronizing data is valid and is up to date. If it is, try rerunning keymap synchronization.\n"
+                        "If the issue persists, you may need to reset this key's keymap data and rerun synchronization again.\n"
+                        "If nothing helps contact data provider.\n\n"
+                        "To re-enable this error, you can set `duplicate_warn_only: false` parameter in keymap configuration.\n"
+                    ),
+                    err=True,
+                )
+            else:
+                raise KeymapDuplicateMapping(key=name, key_count=affected_key_count, affected_count=affected_row_count)
+
+
+@commands.configure.register(Context, RedisKeyMap)
+def configure(context: Context, keymap: RedisKeyMap):
+    rc: RawConfig = context.get("rc")
+
+    keymap.sync_transaction_size = rc.get("keymaps", keymap.name, "sync_transaction_size", default=10000, cast=int)
+    keymap.duplicate_warn_only = rc.get("keymaps", keymap.name, "duplicate_warn_only", cast=bool, default=False)
```

Comment:
Would be nice to keep the keyword parameters in order (any way you want but consistent):

```suggestion
    keymap.sync_transaction_size = rc.get("keymaps", keymap.name, "sync_transaction_size", default=10000, cast=int)
    keymap.duplicate_warn_only = rc.get("keymaps", keymap.name, "duplicate_warn_only", default=False, cast=bool)
```